lib/connection.c

```diff
@@ -512,7 +512,8 @@ STDCALL decorrelate_jitter_init(uint32 base, uint32 cap) {
 
 uint32
 decorrelate_jitter_next_sleep(DECORRELATE_JITTER_BACKOFF *djb, uint32 sleep) {
-    return uimin(djb->cap, uimax(djb->base, (uint32) (rand() % (sleep * 3))));
+    sleep = uimin(sleep, djb->cap);
+    return ((uint32)(sleep/2) + (uint32) (rand() % (sleep/2)));
```

Comment:
Effectively we are doing 
Retry count starts with 0
```
        unsigned long expectedSleepTimeInMs = m_minSleepTimeInMs * pow(2, (m_retryCount+1));
        expectedSleepTimeInMs = (std::min)(expectedSleepTimeInMs, m_maxSleepTimeInMs);
        unsigned long jitterInMs = (unsigned long)(rand() % (expectedSleepTimeInMs/2));
        expectedSleepTimeInMs = (unsigned long)((expectedSleepTimeInMs/2) + jitterInMs);
        return expectedSleepTimeInMs ;
```

Reply:
```
2021-02-06 02:34:58.764 DEBUG C     http_perform.c     Got retryable error http code 503, retry count  1 will retry after 400 micro seconds
2021-02-06 02:34:59.604 retry count  2 will retry after 1400 micro seconds
2021-02-06 02:35:01.129 retry count  3 will retry after 900 micro seconds
2021-02-06 02:35:02.204 retry count  4 will retry after 1100 micro seconds
2021-02-06 02:35:03.451 retry count  5 will retry after 900 micro seconds
2021-02-06 02:35:04.519 retry count  6 will retry after 1500 micro seconds
2021-02-06 02:35:06.184 retry count  7 will retry after 1000 micro seconds
2021-02-06 02:35:07.354 retry count  8 will retry after 1200 micro seconds
2021-02-06 02:35:08.901 retry count  9 will retry after 900 micro seconds
2021-02-06 02:35:09.911 retry count  10 will retry after 1300 micro seconds
2021-02-06 02:35:11.670 retry count  11 will retry after 1000 micro seconds
2021-02-06 02:35:12.812 retry count  12 will retry after 1100 micro seconds
2021-02-06 02:35:14.054 retry count  13 will retry after 1000 micro seconds
2021-02-06 02:35:15.164 retry count  14 will retry after 1100 micro seconds
2021-02-06 02:35:16.404 retry count  15 will retry after 1100 micro seconds
2021-02-06 02:35:17.676 retry count  16 will retry after 1400 micro seconds
2021-02-06 02:35:19.213 retry count  17 will retry after 1200 micro seconds
2021-02-06 02:35:21.112 retry count  18 will retry after 1000 micro seconds
2021-02-06 02:35:22.255 retry count  19 will retry after 1200 micro seconds
2021-02-06 02:35:23.958 retry count  20 will retry after 800 micro seconds
2021-02-06 02:35:25.653 retry count  1 will retry after 500 micro seconds
2021-02-06 02:35:26.312 retry count  2 will retry after 800 micro seconds
2021-02-06 02:35:27.252 retry count  3 will retry after 1500 micro seconds
2021-02-06 02:35:28.915 retry count  4 will retry after 1300 micro seconds
2021-02-06 02:35:30.330 retry count  5 will retry after 1400 micro seconds

```

Reply:
I'm a bit confused, how are we not getting exponential time? We have that `* pow(2, (m_retryCount+1))` which I believe `* 2` should be enough (since `decorrelate_jitter_next_sleep` for an input value `X` return something in range `[X/2, X)`).